include/aspect/boundary_temperature/two_merged_boxes.h

```diff
@@ -0,0 +1,108 @@
+
+#include <aspect/boundary_temperature/interface.h>
+
+
+namespace aspect
+{
+  namespace BoundaryTemperature
+  {
+    /**
+     * A class that implements a temperature boundary condition for a
+     * box geometry with additional boundary indicators for the
+     * lithospheric part of left and right (and front and back in 3D) boundaries.
+     *
+     * @ingroup BoundaryTemperatures
+     */
+    template <int dim>
+    class TwoMergedBoxes : public Interface<dim>
+    {
+      public:
+        /**
+         * Return the temperature that is to hold at a particular location on
+         * the boundary of the domain. This function returns constant
+         * temperatures at the boundaries.
+         *
+         * @param geometry_model The geometry model that describes the domain.
+         * This may be used to determine whether the boundary temperature
+         * model is implemented for this geometry.
+         * @param boundary_indicator The boundary indicator of the part of the
+         * boundary of the domain on which the point is located at which we
+         * are requesting the temperature.
+         * @param location The location of the point at which we ask for the
+         * temperature.
+         */
+        virtual
+        double temperature (const GeometryModel::Interface<dim> &geometry_model,
+                            const types::boundary_id             boundary_indicator,
+                            const Point<dim>                    &location) const;
+
+        /**
+         * Return the minimal the temperature on that part of the boundary on
+         * which Dirichlet conditions are posed.
+         *
+         * This value is used in computing dimensionless numbers such as the
+         * Nusselt number indicating heat flux.
+         */
+        virtual
+        double minimal_temperature (const std::set<types::boundary_id> &fixed_boundary_ids) const;
+
+        /**
+         * Return the maximal the temperature on that part of the boundary on
+         * which Dirichlet conditions are posed.
+         *
+         * This value is used in computing dimensionless numbers such as the
+         * Nusselt number indicating heat flux.
+         */
+        virtual
+        double maximal_temperature (const std::set<types::boundary_id> &fixed_boundary_ids) const;
+
+        /**
+         * Declare the parameters this class takes through input files. This
+         * class declares the inner and outer boundary temperatures.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+        /**
+         * The values of the various temperature variables on each of the
+         * 2*dim+2*(dim-1) boundaries of the box.
+         */
+        double temperature_[2*dim+2*(dim-1)];
```

Comment:
Let's rename this to just `temperature`. We don't use the trailing underscore anywhere else either.